src/Gateway.php

```diff
@@ -108,7 +109,7 @@ public function authorize(array $parameters = []): RequestInterface
      */
     public function completeAuthorize(array $parameters = []): RequestInterface
     {
-        return $this->fetchTransaction($parameters);
+        return $this->createRequest(CompleteAuthoriseAndCaptureRequest::class, $parameters);
```

Comment:
It might make more sense to completeAuthorize & if successful update the status with a fetchTransaction.

Reply:
Why? In the fetchTransaction, you will only look up if we can find a matching transaction in Icepay. Therefore we only look at the status code 200.